### PR TITLE
308 router nats

### LINKED ISSUE #308: [router] Add NATS client
**Describe the bug**
There is no way for the router to broadcast information, like its `config`. A NATS client should be added to the router to allow for this (and other future broadcast information)

### PR DESCRIPTION
**Pull request by LayneHaber**
## The Problem
Fixes #308

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -209,6 +209,7 @@ watch-browser-node: browser-node
 	bash ops/test-unit.sh browser-node watch
 
 test-router: router-js
+	bash ops/start-messaging.sh
```
Comment (rhlsthrm) [inline Makefile:212]: Should we really use real messaging here? Why not the MemoryMessagingService?

File: modules/router/src/services/messaging.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const method = "publishRouterConfig";
+    const methodId = getRandomBytes32();
+    this.logger.debug({ method, methodId, config }, "Method started");
+    const subject = `${this.publicIdentifier}.config`;
+    await this.publish(subject, Result.ok(config).toJson());
+    this.logger.debug({ method, methodId }, "Method complete");
+  }
+}
+
+export const configureSubscriptions = async (
```
Comment (rhlsthrm) [inline modules/router/src/services/messaging.ts:29]: Why is this function necessary?

**Review by rhlsthrm — CHANGES_REQUESTED** [review]
Let's wait on merging this until we get more info about the auctions, it seems like we don't really need this functionality once we have auctions in.

File: Makefile (hunk 1 above)
Comment (LayneHaber) [reply Makefile:212]: Memory messaging service is not really used anywhere, and many of the methods just throw the `Method not implemented` error.

I just copied the setup for the server-node tests which rely on a messaging service being created

File: modules/router/src/services/messaging.ts (hunk 2 above)
Comment (LayneHaber) [reply modules/router/src/services/messaging.ts:29]: We will likely use it in the future to setup other nats subscriptions and broadcasts as we add more functionality to the service, similar to the `setupListeners` function, so I decided to add it in.

**Comment by LayneHaber** [comment]
> Let's wait on merging this until we get more info about the auctions, it seems like we don't really need this functionality once we have auctions in.

Disagree, we've talked a few times about adding in a way to get the router config so you don't send funds to a channel if they won't support your swap

**Pull request merged**